provider/rotation_resource.go

```diff
@@ -0,0 +1,483 @@
+package provider
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/firehydrant/terraform-provider-firehydrant/firehydrant"
+
+	"github.com/hashicorp/go-cty/cty"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceRotation() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: createResourceFireHydrantRotation,
+		ReadContext:   readResourceFireHydrantRotation,
+		UpdateContext: updateResourceFireHydrantRotation,
+		DeleteContext: deleteResourceFireHydrantRotation,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"team_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"schedule_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"time_zone": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"slack_user_group_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"enable_slack_channel_notifications": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+			"prevent_shift_deletion": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+			"coverage_gap_notification_interval": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"color": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"members": {
+				Type:     schema.TypeList,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Optional: true,
+			},
+			"strategy": {
+				Type:     schema.TypeList, // Using TypeList to simulate a map
+				Required: true,
+				ForceNew: true,
+				MaxItems: 1,
```

Comment:
Isn't there a `MinItems` as well?

Reply:
Ah okay! My assumption was that required meant not `nil` and the number of items was another concern.

Good to know!